### ISSUE TITLE
When files are removed from the directory during uploads, the app completely breaks.

### ISSUE THREAD

**Issue by msanterre** [eivindveg/HotSUploader#113]
Since the replay file statuses are kept in a database, if you remove the replay files between the time that it was added as an upload task and the time it was successfully uploaded, you will get constantly get *Connection Error*.

Example of the log:
```
17:28:39.761 [pool-2-thread-1          ] INFO  n.e.h.concurrent.tasks.UploadTask - Uploading replay Garden of Terror (5).StormReplay
17:28:39.761 [JavaFX Application Thread] WARN  n.e.h.models.ReplayFile - Infernal Shrines (3).StormReplay has no statuses.
17:28:39.762 [JavaFX Application Thread] ERROR n.e.h.services.UploaderService - UploadTask failed.
ninja.eivind.mpq.models.MpqException: Could not parse file
	at ninja.eivind.stormparser.StormParser.parseReplay(StormParser.java:33) ~[storm-parser-0.1.1.jar:na]
	at ninja.eivind.hotsreplayuploader.concurrent.tasks.UploadTask.lambda$call$49(UploadTask.java:48) ~[HotS%20Replay%20Uploader-jfx.jar:2.0.3]
	at java.util.Collections$SingletonList.forEach(Collections.java:4822) ~[na:1.8.0_60]
	at ninja.eivind.hotsreplayuploader.concurrent.tasks.UploadTask.call(UploadTask.java:45) ~[HotS%20Replay%20Uploader-jfx.jar:2.0.3]
	at ninja.eivind.hotsreplayuploader.concurrent.tasks.UploadTask.call(UploadTask.java:32) ~[HotS%20Replay%20Uploader-jfx.jar:2.0.3]
	at javafx.concurrent.Task$TaskCallable.call(Task.java:1423) ~[jfxrt.jar:na]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_60]
	at javafx.concurrent.Service.lambda$null$493(Service.java:725) ~[jfxrt.jar:na]
	at java.security.AccessController.doPrivileged(Native Method) ~[na:1.8.0_60]
	at javafx.concurrent.Service.lambda$executeTask$494(Service.java:724) ~[jfxrt.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_60]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_60]
	at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_60]
Caused by: java.io.FileNotFoundException: /Users/moox/Library/Application Support/Blizzard/Heroes of the Storm/Accounts/56880185/1-Hero-1-5311518/Replays/Multiplayer/Garden of Terror (5).StormReplay (No such file or directory)
	at java.io.FileInputStream.open0(Native Method) ~[na:1.8.0_60]
	at java.io.FileInputStream.open(FileInputStream.java:195) ~[na:1.8.0_60]
	at java.io.FileInputStream.<init>(FileInputStream.java:138) ~[na:1.8.0_60]
	at ninja.eivind.mpq.MpqArchive.getFiles(MpqArchive.java:53) ~[mpq-exploder-0.1.1.jar:na]
	at ninja.eivind.stormparser.StormParser.parseReplay(StormParser.java:29) ~[storm-parser-0.1.1.jar:na]
	... 12 common frames omitted
```

Deleting the database file fixes the issue.

**Comment by eivindveg** [comment]
Thank you for a very descriptive bug report. Did you try restarting the application before you removed the database?

**Comment by msanterre** [comment]
@eivindveg I restarted multiple times, also re-installed the app. I eventually figured out through the code where the database file was and deleted it.

**Comment by eivindveg** [comment]
I accidentally uncovered #115 through this issue, which is why the restarts didn't work for purging the file. We'll fix both these issues(a fix for this issue is underway as part of #114), then issue version 2.0.4.

**Comment by eivindveg** [comment]
Fixed in #116
